README.md

```diff
@@ -159,16 +165,16 @@ r.insert(ast::forall(|[p, q, r]| {
     )
     .when(add, vec![p.into(), q.into(), r.into()])
 }));
-```
 
-We can now ask the solver to prove statements within this universe, e.g. that "there exists an X
-such that X + 2 = 3". This statement is indeed true for X = 1, and indeed, the solver will provide
-this answer:
+// We can now ask the solver to prove statements within this universe, e.g. that "there exists an X
+// such that X + 2 = 3". This statement is indeed true for X = 1, and indeed, the solver will provide
+// this answer:
+
+let mut r = RuleResolver::new(&r);
 
-```rust
 // Obtain an iterator that allows us to exhaustively search the solution space:
 let solutions = query_dfs(
-    &r,
+    &mut r,
```

Comment:
This and the following makes `cargo test` pass. Not sure if it's the best was to resolve.